ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rebuild/NonNativeAcidMaterializedViewASTBuilder.java

```diff
@@ -50,12 +50,7 @@ public void appendDeleteSelectNodes(ASTNode selectNode, String tableName) {
 
     for (int i = 0; i < selectNode.getChildCount(); ++i) {
       ASTNode selectExpr = (ASTNode) selectNode.getChild(i);
-      ASTNode expression = (ASTNode) selectExpr.getChild(0);
-      if (expression.getType() == HiveParser.DOT &&
-          expression.getChildCount() == 2 &&
-          expression.getChild(0).getType() == HiveParser.TOK_TABLE_OR_COL) {
-        selectedColumns.add(expression.getChild(1).getText());
-      }
+      selectedColumns.add(selectExpr.getChild(selectExpr.getChildCount() - 1).getText());
```

Comment:
does this fix the same problem for types other than varchar?

Reply:
This issue affect `string` types other than `string`. Those are converted to `string` because of Iceberg type system.

Removed the unused import